### PR TITLE
Lightning workflows moved to monorepo.

### PR DESCRIPTION
**Pull request by vincentmr**
**Context:**
[SC-43411](https://app.shortcut.com/xanaduai/story/43411/update-lightning-devices-in-test-matrix-to-use-new-monorepo) requires compatibility tests between PennyLane and Lightning's most recent stable releases and development versions. 

**Description of the Change:**
This PR removes the Lightning workflows which are now part of Lightning itself (like L-GPU and L-Kokkos). The Lightning status links in README are updated accordingly. 

**Related GitHub issues**
https://github.com/PennyLaneAI/pennylane-lightning/pull/498

### CODE REVIEW

**Review by AmintorDusko — APPROVED** [review]
Hi @vincent, that's a sensible change. Thank you for that.

**Comment by timmysilv** [comment]
darn, my CI workflow that I added didn't account for untracked files. Anyway, could you please remove [lightning from compile.py](https://github.com/PennyLaneAI/plugin-test-matrix/blob/d2a6de89357593cdd7d6b599c25b936dc7b2fc0c/compile.py#L96-L112)?

Also, the base lightning links still point to `plugin-test-matrix` CI actions that you're removing. Can you update those to their new action location?

**Comment by vincentmr** [comment]
> darn, my CI workflow that I added didn't account for untracked files. Anyway, could you please remove [lightning from compile.py](https://github.com/PennyLaneAI/plugin-test-matrix/blob/d2a6de89357593cdd7d6b599c25b936dc7b2fc0c/compile.py#L96-L112)?
> 
> Also, the base lightning links still point to `plugin-test-matrix` CI actions that you're removing. Can you update those to their new action location?

Thanks for that. Links are fixed, I think, but some workflows in Lightning might require an update.

**Review by timmysilv — APPROVED** [review]

File: README.md
[the remark this answers is not in the document]
Comment (vincentmr) [reply README.md:1]: I'm unsure which icons you are referring to. My goal is just to change the statuses from this repo to the new Lightning monorepo.

[the remark this answers is not in the document]
Comment (albi3ro) [reply README.md:1]: I think it was a byproduct of how github was rendering the rich view of a diff. Looking at the branch itself now, I'm realizing that its not there.

Is a github problem, so I'm resolving the conversation.

Screenshot 2023-10-02 at 12 17 24 PM

**Comment by vincentmr** [comment]
@albi3ro Could you please merge this? I cannot as some plugins are failing.

**Comment by vincentmr** [comment]
Thanks @AmintorDusko @albi3ro and @timmysilv

**Pull request merged**